main.go

```diff
@@ -11,22 +12,22 @@ import (
 	"github.com/kong/deck/cmd"
 )
 
-func registerSignalHandler() {
+func registerSignalHandler() context.Context {
+	ctx, cancel := context.WithCancel(context.Background())
 	sigs := make(chan os.Signal, 1)
-	done := make(chan struct{})
-	cmd.SetStopCh(done)
 	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
 
 	go func() {
 		sig := <-sigs
 		fmt.Println("received", sig, ", terminating...")
-		close(done)
+		cancel()
 	}()
+	return ctx
 }
 
 func main() {
-	registerSignalHandler()
-	cmd.Execute()
+	ctx := registerSignalHandler()
```

Comment:
Consider the following:

https://pkg.go.dev/os/signal#NotifyContext

`ctx, _ := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)`

Reply:
included this, this is a handy simplification

Reply:
Said too soon, @mflendrich the only problem here is that when the signal arrives, there wouldn't be a message to stdout stating that the signal was actually received. This is quite helpful in debugging. Thoughts?

Reply:
Hacky way (not recommended):

```go
go func() {
  select {
    case <-ctx.Done():
      fmt.Printf("context cancelled (%w), terminating...", ctx.Err())
  }
}
```

Clean way:
- Replace `os.Exit(1)` throughout the codebase with errors plumbed all the way through to `RunE` and to the return value of `cmd.Execute()`
- After `cmd.Execute()` returns, check if `ctx.Err()` is not nil. If it isn't, then the context got cancelled by the signal.

Both are somewhat tedious and may defeat the purpose of this simplification.

Reply:
Thought and explored a few different ways to do it but I haven't found a good way to deal with it in this PR.
This also feels minor enough that we can leave it be for now. If it is a bigger issue, please LMK and I'll address with a quick follow up PR.
I'm going to go ahead and merge this in.